server/src/controllers/Events/resolver.ts

```diff
@@ -230,44 +229,36 @@ const getNameForNewAttendance = (event: EventAttendanceName) => {
 
 @Resolver()
 export class EventResolver {
-  @Query(() => PaginatedEventsWithTotal)
+  @Query(() => [PaginatedEventsWithChapters])
   async paginatedEventsWithTotal(
     @Arg('limit', () => Int, { nullable: true }) limit?: number,
     @Arg('offset', () => Int, { nullable: true }) offset?: number,
-  ): Promise<PaginatedEventsWithTotal> {
-    const total = await prisma.events.count();
-    const events = await prisma.events.findMany({
-      include: {
-        chapter: true,
-      },
-      orderBy: [{ start_at: 'asc' }, { name: 'asc' }],
-      take: limit ?? 10,
-      skip: offset,
+    @Arg('showOnlyUpcoming', () => Boolean, { nullable: true })
+    showOnlyUpcoming = true,
+  ): Promise<PaginatedEventsWithChapters[]> {
+    const total = await prisma.events.count({
+      ...(showOnlyUpcoming && {
+        where: {
+          canceled: false,
+          ends_at: { gt: new Date() },
+        },
+      }),
     });
-    return { total, events };
-  }
-
-  @Query(() => [EventWithChapter])
-  async paginatedEvents(
-    @Arg('limit', () => Int, { nullable: true }) limit?: number,
-    @Arg('offset', () => Int, { nullable: true }) offset?: number,
-  ): Promise<EventWithChapter[]> {
-    return await prisma.events.findMany({
-      where: {
-        AND: [
-          {
-            canceled: false,
-            ends_at: { gt: new Date() },
-          },
-        ],
-      },
+    const events = await prisma.events.findMany({
+      ...(showOnlyUpcoming && {
+        where: {
+          canceled: false,
+          ends_at: { gt: new Date() },
+        },
+      }),
       include: {
         chapter: true,
       },
       orderBy: [{ start_at: 'asc' }, { name: 'asc' }],
       take: limit ?? 10,
       skip: offset,
     });
+    return events.map((event) => ({ ...event, total }));
```

Comment:
Adding `total` to each event object might be reason for part of the issues. Why not return `total` separately?